cmake/configs/modes.cmake

```diff
@@ -130,6 +130,9 @@ else()
     endif()
 endif()
 
+# Address sanitizer builds should have debug info available and optimization off
+set(san_debug_flags "-O0 -g ${no_omit_frame_flag}")
```

Comment:
Same comment about `-Og`, but do we also need to force `-g` here? For some reason I thought the sanitiser builds already used `-g` (via `HAVE_DIAGNOSTIC`) but I could be wrong.

Reply:
I agree that the HAVE_DIAGNOSTIC code turns `-g` on for some cases, including this at the moment. It seems to me like a tenuous connection to link a compile time define in WiredTiger with build flags.

I'm not sure how big to make this change, so I didn't want to go messing with that code right here. I'd be OK to expand the scope, but I think that would require a more careful review of evergreen.yml and it's usage of HAVE_DIAGNOSTIC to make sure we got things right.

FWIW: I included this here to (hopefully) match the default cmake_build_type switch as well. I'm presuming that debug builds will include disabling optimizations and outputting debug information.

Reply:
> It seems to me like a tenuous connection to link a compile time define in WiredTiger with build flags.

That's fair. If the goal is to guarantee a particular optimisation level and debugging flags for sanitiser builds no matter what, then this is the way to do it. One thing to be aware of here is that these flags won't be override-able from the command line since they're the final occurrence of these options - not sure how much we care about that for sanitiser builds.

Reply:
I considered making explicit release and debug versions of these build types, but figured it wasn't necessary at the moment. I'm not convinced that overriding the build_type is the right approach to Sanitizer builds moving forward, so I'm inclined to take this, and resolve the generalization in the future when we tidy up the build/test setup more broadly.